server/api/controllers/webhook/getData.ts

```diff
@@ -0,0 +1,25 @@
+import { firestore } from './firebase';
+import { collection, query, getDocs } from '@firebase/firestore';
+const ref = collection(firestore, 'clients');
+type Record = { board_ID: string; username: string; api_key: string };
+const allData: Record[] = [];
+
+const getData = async () => {
+  const q = query(ref);
+
+  const querySnapshot = await getDocs(q);
+  querySnapshot.forEach((doc) => {
```

Comment:
prefer using `.map` over `.forEach`

Reply:
firebase does not allow usage of `.map` but `.forEach`